### ISSUE TITLE
Bernoulli trial behavior in Tweek

### ISSUE THREAD

**Issue by Yshayy** [Soluto/tweek#707]
Currently, gradual release behaves in the following way - for 0.1 value split the population to two groups 
a. 1-10 - TRUE
b. 11-100 - FALSE
because group b got an explicit value, Tweek will not move to the next rule for this group.
It might be more correct to translate group B value to NONE, which means no match was found and evaluation will continue to the next rule, from semantic point of view, it means we no longer split the population to two groups, but rather add a condition which is like "user is in x% percentile".
After this change, gradual release behaves in a bit "greedier" way.
It can be relevant if we want the users to have more opportunities to get 'true' value.
It can also be relevant for time based releases:
For Key A:
1. CurrentDate greater then 01/01/2018 open feature to 10% of users.
2. CurrentDate greater then 08/01/2018 open feature to 30% of users.
Assuming theses rules share the same salt, this behavior will behave correct after this change, currently, we will need to add additional constraint for the first rule (date lower than 08/01/2018)  for that to work.

Suggested change:
1. For rules with gradual release, change Bernoulli trial behavior to use TRUE-NONE instead of TRUE-FALSE, allowing the system to be 'greedy' and move to the next rule.
2. When creating a new bool key, set default value to FALSE, to avoid null behavior of key when using gradual release.

**Comment by nataly87s** [comment]
i think there should be an option to select if we want to continue with the rules or return false

**Comment by Yshayy** [comment]
As another parameter for Bernoulli? What should be the default?

**Comment by nataly87s** [comment]
Yes, another parameter. The default should be TRUE-FALSE, for backwards compatibility
